src/main/java/seedu/address/model/event/UidList.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import static javafx.collections.FXCollections.observableArrayList;
+import static javafx.collections.FXCollections.unmodifiableObservableList;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Uid;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * A list of uids of persons.
+ * */
+public class UidList implements Iterable<Uid> {
+    private final ObservableList<Uid> internalList = observableArrayList();
+    private final ObservableList<Uid> internalUnmodifiableList =
+            unmodifiableObservableList(internalList);
+    private String personNames = "";
+    /**
+     * Returns true if the uid to be checked is already present in the uid list.
+     * @param toCheck uid to be checked if a duplicate is present in the list.
+     */
+    public boolean contains(Uid toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameUid);
+    }
+    /**
+     * Adds a new uid to the list of uids it stores in its list.
+     *
+     * @param toAdd uid to be added to the list.
+     */
+    public void add(Uid toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicatePersonException();
+            // this operation will result in Duplicates Persons.
+        }
+        internalList.add(toAdd);
+    }
+    /**
+     * Removes the equivalent person from the list.
```

Comment:
Perhaps rephrase this for clarity

```suggestion
     * Removes the person associated with the given {@code Uid toRemove} from the list.
```